File: py/0523-continuous-subarray-sum.py
class Solution:
    def checkSubarraySum(self, nums: list[int], k: int) -> bool:
        totals = {0: -1}
        total = 0
        for i, n in enumerate(nums):
            # k != 0 check not needed
            total = (total + n) % k
            if total in totals:
                if i - totals[total] > 1:
                    return True
            else:
                totals[total] = i
        return False

File: cpp/0523-continuous-subarray-sum.cpp
#include <unordered_map>
#include <vector>

class Solution {
    public:
        [[nodiscard]] auto
        checkSubarraySum(const std::vector<int> &nums, const int k) -> bool {
            auto totals = std::unordered_map<int, int>{{0, -1}};
            auto total = 0;
            for (auto i = 0; const auto n : nums) {
                // k != 0 check not needed
                total = (total + n) % k;
                const auto it = totals.insert({total, i}).first;
                if (i - it->second > 1) {
                    return true;
                }
                ++i;
            }
            return false;
        }
};
